### ISSUE TITLE
Error uploading JUnit XML Artifact

### ISSUE THREAD

**Issue by loskylp** [gocd/gocd#2336]
##### Summary

I have a pipeline and in a test stage it generates Junit Report as XML (tryed 1 XML per Test and All In One). On upload the go-server trasforms XML -RDF and then fails to parse the generated 
##### Environment

http://<go-server-host>/api/go/support. wasn't valid url
http://<go-server-host>/go/api/support was

```
Timestamp:
==========
Thu Jun 23 04:56:16 UTC 2016
=================
Go Server Version
=================

16.5.0(3305-5f54cf4338eb2ff5c9f486ede6ae97e9aae51e82
==============
OS information
==============

Linux, amd64, 4.2.8-1-pve, 4, 0.05
===================
Runtime information
===================

Name - 1377@gocd
Uptime - 123530589 [About 34 hours, 18 minutes, 50 seconds]
Spec Name - Java Virtual Machine Specification
Spec Vendor - Oracle Corporation
Spec Version - 1.7

Input Arguments
---------------
  -> -Djava.security.egd=file:/dev/./urandom
  -> -Xms512m
  -> -Xmx1024m
  -> -XX:PermSize=128m
  -> -XX:MaxPermSize=256m
  -> -Duser.language=en
  -> -Djruby.rack.request.size.threshold.bytes=30000000
  -> -Duser.country=US
  -> -Dcruise.config.dir=/etc/go
  -> -Dcruise.config.file=/etc/go/cruise-config.xml
  -> -Dcruise.server.port=8153
  -> -Dcruise.server.ssl.port=8154
```
###### Basic environment details
- Go Version: 16.5.0(3305-5f54cf4338eb2ff5c9f486ede6ae97e9aae51e82)
- JAVA Version:
  java.vendor.url=http://java.oracle.com/
  java.version=1.7.0_101
  java.vm.version=24.95-b01
- OS: Linux, amd64, 4.2.8-1-pve, 4, 0.05
##### Steps to Reproduce
1.  Pipeline with Test Stage
2.  Build and get JUnit results in XML
3.  put the /resport/junit dir as Artifact
4.  run and fail tests
##### Expected Results

The test resports should appear in the test tab 
##### Actual Results

The tab shows as follows:

```
Tests run: 0 , Failures: 0 , Not run: 0 , Time: .000 seconds.

No Tests Run
This project doesn't have any tests
```
##### Log snippets

```
2016-06-23 00:32:32,970 ERROR [qtp1250610188-28] GRDDLTransformer:55 - Could not convert to a graph. The document was:
<?xml version="1.0" encoding="UTF-8"?>
<rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:xunit="http://studios.thoughtworks.com/ontologies/2010/03/24-xunit#" xmlns:java="http://xml.apache.org/xalan/java">
      <xunit:hasFailure><xunit:Failure rdf:about="urn:uuid:f2d19bdd-371d-4733-8e4f-40b68efa6b5f"><xunit:failureMessage rdf:datatype="http://www.w3.org/2001/XMLSchema#string">com.loskysoft.app.TestOperacionFactory</xunit:failureMessage><xunit:failureType rdf:datatype="http://www.w3.org/2001/XMLSchema#string">java.lang.ClassNotFoundException</xunit:failureType><xunit:failureStackTrace rdf:datatype="http://www.w3.org/2001/XMLSchema#string">java.lang.ClassNotFoundException: com.loskysoft.app.TestOperacionFactory
    at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
    at java.security.AccessController.doPrivileged(Native Method)
.
.
.

com.thoughtworks.studios.shine.ShineRuntimeException: Could not create graph from XML RDF!
    at com.thoughtworks.studios.shine.semweb.sesame.SesameGraph.addTriplesFromRDFXMLAbbrev(SesameGraph.java:285)
    at com.thoughtworks.studios.shine.semweb.grddl.GRDDLTransformer.transform(GRDDLTransformer.java:52)
    at com.thoughtworks.studios.shine.xunit.AntJUnitReportRDFizer.importXUnit(AntJUnitReportRDFizer.java:62)
    at com.thoughtworks.studios.shine.xunit.AntJUnitReportRDFizer.importFile(AntJUnitReportRDFizer.java:57)
    at com.thoughtworks.studios.shine.cruise.stage.details.XMLArtifactImporter.importXML(XMLArtifactImporter.java:72)

```
##### Any other info

Cant attach XML so here is a snipset

```
<?xml version="1.0" encoding="UTF-8" ?>
<testsuites>
  <testsuite errors="0" failures="0" hostname="solaria" id="0" name="TestOperacionFactory" package="com.loskysoft.app" skipped="0" tests="2" time="0.885" timestamp="2016-06-23T00:15:40">
      <properties>
.
.
.
     </properties>
      <testcase classname="com.loskysoft.app.TestOperacionFactory" name="testObtenerOperacion" time="0.601" />
      <testcase classname="com.loskysoft.app.TestOperacionFactory" name="testFallarObtenerOperacion" time="0.003" />
      <system-out><![CDATA[]]></system-out>
      <system-err><![CDATA[]]></system-err>
  </testsuite>
  <testsuite errors="0" failures="0" hostname="solaria" id="1" name="TestTaskInformationService" package="com.loskysoft.app" skipped="0" tests="4" time="0.356" timestamp="2016-06-23T00:15:42">
      <properties>

```

**Comment by ketan** [comment]
Can you put the XML onto gist.github.com and link it here. I'd be really useful to have a copy of the XML without which we can't fix the issue.

**Comment by loskylp** [comment]
Here is the gist
https://gist.github.com/loskylp/c258eae82a11cb926b7cbf3e0df1c525

**Comment by ggavrilov** [comment]
I have the same issue. For my project it is killer feature of CI/CD server. Can you get answer when this bug planing to fix or will not fix? I need choose the CI server and Go fits but this bug is show stopper.

**Comment by zabil** [comment]
@ggavrilov 
Can you share your logs? 
I doubt it's the same error. Have you marked your junit results as a test artifact?
More details on how to do that here
https://docs.go.cd/current/configuration/dev_upload_test_report.html

**Comment by PredatorVI** [comment]
I think I saw this and fixed it for my environment.  Though the core issue might still be a bug.

I have 2 agents this constantly happened on and it is causing severe disruption in our process because all agent processes will "LostContact" and no pipelines will process.  All agents and server are running the latest version as of 3/11/2019 (v19.2.0-8641)

On the affected agents, there are no errors that appear related to go-agent processes.  All logs that I've looked at (go-shine.log) show that a maven test run failed and is uploading the logs:

`2019-03-08 12:32:55,309 WARN  [scheduler-2] BaseCommandBuilder:92 - return code is 1
2019-03-08 12:32:55,349 INFO  [scheduler-2] HttpService:138 - Got back 200 from server
2019-03-08 12:32:55,349 INFO  [scheduler-2] DefaultGoPublisher:101 - Agent [go-agent06.mycompany.com, 10.131.24.205, 8fad4457-695b-45d5-9af8-f1eb9ff9a397] is reporting build result [Failed] to Go Server for Build [responders_INT/46/Validate/1/validate/28411]
2019-03-08 12:32:55,381 INFO  [scheduler-2] HttpService:138 - Got back 200 from server
2019-03-08 12:32:55,382 INFO  [scheduler-2] DefaultGoPublisher:95 - Agent [go-agent06.mycompany.com, 10.131.24.205, 8fad4457-695b-45d5-9af8-f1eb9ff9a397] is reporting status [Completing] to Go Server for Build [responders_INT/46/Validate/1/validate/28411]
2019-03-08 12:32:55,410 INFO  [scheduler-2] ArtifactsPublisher:69 - Pluggable metadata folder is empty.
2019-03-08 12:32:57,641 INFO  [scheduler-2] HttpService:71 - Uploading file [/var/lib/go-agent/data/cruise-5730ea53-05c1-4215-a110-b76c17960317/9efb8cf5-8076-4c6b-b75d-67ccf3891a16/surefire-reports.zip] to url [https://go.mycompany.com:8154/go/remoting/files/responders_INT/46/Validate/1/validate/responders_results/qa-libs/qa-responders/target?attempt=1&buildId=28411]
2019-03-08 12:32:58,768 INFO  [scheduler-2] HttpService:138 - Got back 201 from server
2019-03-08 12:32:59,008 INFO  [pool-4-thread-1] HttpService:138 - Got back 200 from server`

On the server side, I see lots of these:

`com.thoughtworks.studios.shine.ShineRuntimeException: Could not create graph from XML RDF!
        at com.thoughtworks.studios.shine.semweb.sesame.SesameGraph.addTriplesFromRDFXMLAbbrev(SesameGraph.java:287)
        at com.thoughtworks.studios.shine.semweb.grddl.GRDDLTransformer.transform(GRDDLTransformer.java:57)
        at com.thoughtworks.studios.shine.xunit.AntJUnitReportRDFizer.importXUnit(AntJUnitReportRDFizer.java:57)
        at com.thoughtworks.studios.shine.xunit.AntJUnitReportRDFizer.importFile(AntJUnitReportRDFizer.java:52)
        at com.thoughtworks.studios.shine.cruise.stage.details.XMLArtifactImporter.importXML(XMLArtifactImporter.java:71)
     ...
Caused by: org.eclipse.rdf4j.rio.RDFParseException: unexpected literal [line 34, column 20]
        at org.eclipse.rdf4j.rio.helpers.RDFParserHelper.reportError(RDFParserHelper.java:322)
        at org.eclipse.rdf4j.rio.helpers.AbstractRDFParser.reportError(AbstractRDFParser.java:684)
        at org.eclipse.rdf4j.rio.rdfxml.RDFXMLParser.reportError(RDFXMLParser.java:1237)
        at org.eclipse.rdf4j.rio.rdfxml.RDFXMLParser.text(RDFXMLParser.java:563)`

As I dug into it, I had one pipeline running a test that had a **typo in the path** of the test files/folders it was trying to archive/copy as a test artifact.  After fixing this, it seems to have helped.  I'm still getting and occasional off-line for another reason but maybe this will help others.
